app/lib/saml_metadata.rb

```diff
@@ -0,0 +1,34 @@
+# frozen_string_literal: true
+
+require 'nokogiri'
+require 'open-uri'
+
+class SAMLMetadata
+  def self.metadata_for(entity_id)
+    # make sure this is a parseable URI, then URI escape it to avoid path
+    # traversal / query injection
+    safe_entity_id = URI.encode_www_form_component(URI(entity_id).to_s)
+    URI.open("#{Otis.config.met_entity_endpoint}/#{safe_entity_id}?viewxml=true")
+  end
+
+  def initialize(entity_id, data: SAMLMetadata.metadata_for(entity_id))
+    @entity_id = entity_id
+    @doc = Nokogiri::XML(data)
+  end
+
+  def name
+    @doc.xpath('//mdui:DisplayName|//md:organizationDisplayName|//md:OrganizationName').first.text
+  end
+
+  def scopes
+    @doc.xpath('//md:IDPSSODescriptor/md:Extensions/shibmd:Scope').map(&:text)
```

Comment:
Seeing Nokogiri::XML::XPath::SyntaxError in HTInstitutionsController#new when I drop in various randomly-selected entity IDs here. For example https://shibboleth.fhwn.ac.at/idp/shibboleth

Reply:
I'll take a look -- thanks!

Reply:
Unfortunately met.refeds.org seems to be down right now, but I suspect this is a namespace issue and that I know how to fix it.

Reply:
I pushed a fix that should address this.